### DISCUSSION TITLE
Proof of concept demo : Smart contract example with Bevy

### DISCUSSION THREAD

**Discussion by pyweeker** [bevyengine/bevy#4030] [Ideas]
Hello,
i 'd like an example of bevy ,out of gamedev.
i thought ECS matches with contracts. Solana cryptomoney is written with rust, seems in rising dynamic , so , any idea if Bevy is adapted for this job.

Maybe we can imagine a wining bet with international soccer championship ?

**Comment by nicopap** [comment]
A few things:

1. Betting is a highly regulated field. Sport betting in particular. What you are suggesting is **highly illegal** in many countries. Adding such an example would incriminate the bevy contributors.

2. There is nothing magical about smart contracts that makes interacting with them different from interacting with any other libraries. The rust ecosystem has many high quality crates that deal with blockchains (specifically Polka as you are suggesting) you should use one of them! Then integration is just like any other libraries. You can already find many bevy plugins that deals with non-bevy crates to integrate them into bevy. See [bevy_kira_audio](https://github.com/NiklasEi/bevy_kira_audio/). The bevy community focuses on general gaming and scientific simulation use cases. If you are interested in niche stuff, you are free to peruse the examples relevant to the bevy community and interpolate for your own use case.

3. Specifically picking one blockchain over another is not the job of the bevy community. It seems your suggestion might be motivated specifically in spreading the good word about one of them. Bevy wouldn't make friend from specifically selecting one over another.

4. The gaming community and a lot of game dev has a profound hatred for cryptocurrencies and smart contracts. Whether this is warranted or not is not for me to decide. However, this means even providing an example using smart contracts would antagonise a lot of the target audience of bevy and be extremely detrimental for the project. This is true regardless of whether personally you think smart contracts are a good thing or not.

**Comment by pyweeker** [comment]
Hello nicopap, thanks for your answer, i have collected 17 down vote before getting an explanation.

**Reply by james7132** [reply]
I think that speaks to how viscerally both the gaming and game dev community responds to cryptocurrency related topics.

It seems like the community has struck a consensus regarding crypto in Bevy proper. However, you're free to pursue this on your own time in your own repo.
